packages/webpack-cli/lib/utils/package-exists.js

```diff
@@ -1,9 +1,14 @@
+const childProcess = require('child_process');
+
+// Node.js has cache on `require.resolve` and second `require` throws the error with `MODULE_NOT_FOUND` code
 function packageExists(packageName) {
-    try {
-        return require.resolve(packageName);
-    } catch (error) {
-        return false;
-    }
+    const nodePath = process.execPath;
+    const nodeOptions = process.execArgv.filter((arg) => !arg.startsWith('--inspect'));
+    const command = `require.resolve(${JSON.stringify(packageName)})`;
+
+    const result = childProcess.spawnSync(nodePath, [...nodeOptions, '-e', command], { encoding: 'utf8', maxBuffer: 1000 * 1000 * 100 });
+
+    return result.status === 0;
```

Comment:
I think we don't need check error message/path/something else, if it fail we should always try to install, even if something goes wrong we will catch it in `require` anyway

Reply:
Yeah you are right. This seems much better check 😄

Reply:
Here one bottleneck - perf, before it was fast, no it is not fast due external process, maybe we can cache result, need time to think

Reply:
Yeah, I also think this isn't the best solution performance-wise.

You can create a very simple lookup solution:

``` js
let dir = __dirname;
do {
  try {
    if(fs.statSync(path.join(dir, "node_modules", packageName)).isDirectory()) return true;
  } catch(e) { }
} while(dir !== (dir = path.dirname(dir)));
return false;
```